intake_xarray/xzarr.py

```diff
@@ -25,12 +25,10 @@ def __init__(self, urlpath, storage_options=None, metadata=None, **kwargs):
 
     def _open_dataset(self):
         import xarray as xr
-        from fsspec import get_mapper
-
-        self._mapper = get_mapper(self.urlpath, **self.storage_options)
-        self._ds = xr.open_zarr(self._mapper, **self.kwargs)
+        self._ds = xr.open_dataset(self.urlpath, engine='zarr',
+                                   backend_kwargs={"storage_options": self.storage_options},
+                                   **self.kwargs)
```

Comment:
Could we add in `xr.open_mfdataset` in here, or do it in a separate PR?

```suggestion
        if "*" in url or isinstance(url, list):
            _open_dataset = xr.open_mfdataset
        else:
            _open_dataset = xr.open_dataset
        self._ds = _open_dataset(self.urlpath, engine='zarr',
                                 backend_kwargs={"storage_options": self.storage_options},
                                 **self.kwargs)
```

Probably need to include more code following the NetCDF driver at https://github.com/intake/intake-xarray/blob/bd43095c3f418aaf6f5cda0d3fbdf108243611e7/intake_xarray/netcdf.py#L65-L76

Reply:
Good idea, but let's make it a separate follow-on PR (since the zarr backend didn't support multiple datasets before anyway). That code looks like it doesn't have anything specific to zarr, so it should be reusable.

Reply:
Sounds good :+1:

Reply:
Yes, I do have something kind of working. @martindurant was working on doing some changes at the xarray level that would have changed the way multiple zarr files are passed through intake. I see that that pull request hasn't passed... I guess I am just waiting on the new implemented changes and to see how tests are going to be changing.

Reply:
Should be close, that PR at https://github.com/pydata/xarray/pull/4461 is approved, just needs to get merged